tests/unit/action/test_bonsai_asset.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import pytest
+
+from ansible.playbook.task import Task
+
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    bonsai, errors,
+)
+from ansible_collections.sensu.sensu_go.plugins.action import bonsai_asset
+
+
+class TestValidate:
+    @pytest.mark.parametrize("name,args,required,typ", [
+        # Required values must match the selected type.
+        ("a", dict(a=3), True, int),
+        ("a", dict(a=3.3), True, float),
+        ("a", dict(a="b"), True, str),
+        ("a", dict(a=[]), True, list),
+        ("a", dict(a={}), True, dict),
+        # Optional values are not checked for type-correctness.
```

Comment:
I find this comment a bit confusing. Based on the comment, I was expecting to see cases like:
 ```
("a", dict(a="b"), False, int) #  optional parameter "a" is given and has wrong type
```
that would pass. But cases like the one above fail (as they should). I'm not sure what you meant here.

Perhaps we need another set of cases to ensure that we have optional parameters which are provided and of the correct type, like this:
```
("a", dict(a=1), False, int)
("a", dict(a=1.2), False, float)
("a", dict(a="b"), False, str)
# ...
```

Reply:
What I meant to say was `# Optional values are not checked for type-correctness if they are missing.`. And now that I am thinking about this, I also forgot to include a test to make sure optional value fails the validation if the type is not correct.